src/lib.rs

```diff
@@ -583,126 +893,190 @@ impl CreateBufferMapped<'_> {
 
     /// Unmaps the buffer from host memory and returns a [`Buffer`].
     pub fn finish(self) -> Buffer {
-        backend::device_create_buffer_mapped_finish(self)
-    }
-}
-
-impl Surface {
-    /// Creates a surface from a raw window handle.
-    pub fn create<W: raw_window_handle::HasRawWindowHandle>(window: &W) -> Self {
-        Surface {
-            id: backend::device_create_surface(window),
+        <C as Context>::flush_mapped_data(self.mapped_data, self.detail);
+        Context::buffer_unmap(&*self.context, &self.id);
+        Buffer {
+            context: self.context,
+            id: self.id,
+            //detail: self.detail,
         }
     }
+}
 
-    #[cfg(any(target_os = "ios", target_os = "macos"))]
-    pub fn create_surface_from_core_animation_layer(layer: *mut std::ffi::c_void) -> Self {
-        Surface {
-            id: wgn::wgpu_create_surface_from_metal_layer(layer),
+impl Instance {
+    /// Create an new instance.
+    pub fn new() -> Self {
+        Instance {
+            context: Arc::new(C::init()),
         }
     }
-}
 
-impl Adapter {
     /// Retrieves all available [`Adapter`]s that match the given backends.
     #[cfg(not(target_arch = "wasm32"))]
-    pub fn enumerate(backends: BackendBit) -> Vec<Self> {
-        wgn::wgpu_enumerate_adapters(backends)
+    pub fn enumerate_adapters(&self, backends: wgt::BackendBit) -> Vec<Adapter> {
```

Comment:
Thoughts on returning an iterator here instead of collecting?

Reply:
Sounds good!